oozie-to-airflow/mappers/mapreduce_mapper.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Maps Oozie pig node to Airflow's DAG"""
+import os
+from typing import Set, Dict
+from xml.etree.ElementTree import Element
+
+import jinja2
+from airflow.utils.trigger_rule import TriggerRule
+
+from definitions import ROOT_DIR
+from mappers.action_mapper import ActionMapper
+from mappers.prepare_mixin import PrepareMixin
+from utils import el_utils, xml_utils
+from utils.file_archive_extractors import FileExtractor, ArchiveExtractor
+
+
+# pylint: disable=too-many-instance-attributes
+class MapReduceMapper(ActionMapper, PrepareMixin):
+    """
+    Converts a MapReduce Oozie node to an Airflow task.
+    """
+
+    properties: Dict[str, str]
+    params_dict: Dict[str, str]
+
+    def __init__(
+        self,
+        oozie_node: Element,
+        name: str,
+        trigger_rule: str = TriggerRule.ALL_SUCCESS,
+        params=None,
+        template_file_name: str = "mapreduce.tpl",
+        **kwargs,
+    ):
+        ActionMapper.__init__(self, oozie_node=oozie_node, name=name, trigger_rule=trigger_rule, **kwargs)
+        if params is None:
+            params = dict()
+        self.template = template_file_name
+        self.params = params
+        self.task_id = name
+        self.trigger_rule = trigger_rule
+        self.properties = {}
+        self.params_dict = {}
+        self.file_extractor = FileExtractor(oozie_node=oozie_node, params=params)
+        self.archive_extractor = ArchiveExtractor(oozie_node=oozie_node, params=params)
+        self._parse_oozie_node()
+
+    def _parse_oozie_node(self):
+        name_node_text = self.oozie_node.find("name-node").text
+        self.name_node = el_utils.replace_el_with_var(name_node_text, params=self.params, quote=False)
```

Comment:
Where is this field used? I can not find use of this field

Reply:
I think that you should pass it as a property ``core:fs.defaultFS={self.name_node}``

Reply:
Pass it where? Can you provide any link describing this?

Reply:
After an in-person discussion we've decided to abandon this as it's not currently possible to pass such a claster configuration option the the DAG.

Reply:
Reference: https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/cluster-properties